frontend/packages/ceph-storage-plugin/src/components/modals/volume-snapshot-modal/volume-snapshot-modal.tsx

```diff
@@ -8,56 +8,253 @@ import {
 
 export type VolumeSnapshotModalProps = {
   pvcData?: FirehoseResourcesResult;
 } & HandlePromiseProps &
   ModalComponentProps;
 
+enum NumberOfSnapshot {
+  THREE = '3',
+  FIVE = '5',
+  SEVEN = '7',
+}
+
 export const VolumeSnapshotModal = withHandlePromise((props: VolumeSnapshotModalProps) => {
   const { close, cancel, pvcData, errorMessage, inProgress, handlePromise } = props;
   const resource = pvcData.data as K8sResourceKind;
+  const [snapshotCount, setSnapshotCount] = React.useState(); // For all schedules
+  const [snapshotWeek, setSnapshotWeek] = React.useState('0'); // For Weekly schedule
+  const [snapshotTime, setSnapshotTime] = React.useState('00:01'); // For weekly and monthly
+  const [snapshotMonth, setSnapshotMonth] = React.useState('1'); // For monthly schedule
   const [snapshotName, setSnapshotName] = React.useState(
     `${getName(resource) || 'pvc'}-snapshot-1`,
   );
+  const [scheduleLabel, setScheduleLabel] = React.useState('');
+  const [scheduleType, setScheduleType] = React.useState('SingleSnapshot');
+
   const snapshotTypes = {
-    single: 'Single Snapshot',
+    SingleSnapshot: 'SingleSnapshot',
+    CronJob: 'CronJob',
+    MonthlySnapshot: 'MonthlySnapshot',
+    WeeklySnapshot: 'WeeklySnapshot',
+  };
+
+  const selectedScheduleType = snapshotTypes.SingleSnapshot;
+
+  const makeLabel = () => {
+    let newLabel = '';
+    const arr = snapshotTime.split(':');
+    newLabel = `${arr[1]} ${arr[0]} `;
+    if (scheduleType === 'MonthlySnapshot') {
+      newLabel = `${newLabel + snapshotMonth} * *`;
+    } else if (scheduleType === 'WeeklySnapshot') {
+      newLabel = `${newLabel}* * ${snapshotWeek}`;
+    }
+    return newLabel;
+  };
+
+  const getWeeklyElements = () => {
+    return (
+      <FormGroup
+        className="ceph-volume-snapshot-modal--input"
+        fieldId="snapshot-modal__day"
+        label="Day of week"
+      >
+        <TextInput
+          type="number"
+          name="snapshot-modal__day"
+          aria-label="snapshot-modal__day"
+          value={snapshotWeek}
+          onChange={setSnapshotWeek}
+          className="ceph-volume-snapshot-modal--label"
+          placeholder="0"
+        />
+      </FormGroup>
+    );
+  };
+
+  const getMonthlyElements = () => {
+    return (
+      <FormGroup
+        className="ceph-volume-snapshot-modal--input"
+        fieldId="snapshot-modal__month"
+        label="Day of month"
+      >
+        <TextInput
+          type="number"
+          name="snapshot-modal__month"
+          aria-label="snapshot-modal__month"
+          value={snapshotMonth}
+          onChange={setSnapshotMonth}
+          className="ceph-volume-snapshot-modal--label"
+          placeholder="1"
+        />
+      </FormGroup>
+    );
+  };
+
+  const getCronElements = () => {
+    return (
+      <FormGroup
+        className="ceph-volume-snapshot-modal--input"
+        fieldId="snapshot-modal__schedule"
+        helperText="* Scheduled for UTC+3"
```

Comment:
maybe use getTimezoneOffset?

Reply:
Have changed it to `Scheduled for cluster local time` because we shouldn't display the browser's time.